src/Sulu/Bundle/AdminBundle/Resources/js/views/Datagrid/Datagrid.js

```diff
@@ -82,6 +85,37 @@ class Datagrid extends React.Component<ViewProps> {
         router.bind('limit', this.datagridStore.limit, 10);
     }
 
+    @action componentDidMount() {
+        const {router} = this.props;
+        const {
+            route: {
+                options: {
+                    locales,
+                    toolbarActions,
+                },
+            },
+        } = router;
+
+        if (!toolbarActions) {
+            return;
+        }
+
+        this.toolbarActions = toolbarActions.map((toolbarAction) => new (toolbarActionRegistry.get(toolbarAction))(
+            this.datagridStore,
+            this,
+            router,
+            locales
+        ));
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        if (!equals(this.props.route.options.locales, prevProps.route.options.locales)) {
```

Comment:
this id different to Form.js not sure if this is the correct way to get the locales